### ISSUE TITLE
Having some trouble

### ISSUE THREAD

**Issue by acomito** [puranjayjain/react-materialui-notifications#76]
I'm running into a few issues. Have you tested the package since upgrading to the new material-ui?

https://github.com/callemall/material-ui/issues/5573

**Comment by puranjayjain** [comment]
@acomito not yet, was planning to launch it with the last remaining feature added.

**Comment by acomito** [comment]
Okay. If I come up with anything helpful I'll report back. Great looking package even if I can't get it working just yet!

**Comment by puranjayjain** [comment]
@acomito haha thank you. Hope you get it working soon. Feel free to ask again if you have any questions or problems.

**Comment by acomito** [comment]
Tried to use this package again today but no luck. So are the steps to integrating it with a material-ui project anything more than installing from npm and calling:

```
ReactMaterialUiNotifications.showNotification({
      title: 'Title',
      additionalText: `Some message to be displayed`,
      overflowText: "joe@gmail.com",
      icon: <Message />,
    });
```

Or do I need to put the `<ReactMaterialUiNotifications />` component in my root component?

I tried both ways and am not seeing anything (no error message either)

**Comment by puranjayjain** [comment]
Yes you need to call it and also put that in the react tree preferably as the root component or somewhere else  in the tree.

**Comment by puranjayjain** [comment]
See this for an implemention example https://github.com/puranjayjain/react-materialui-notifications/blob/master/src/app/Main.js

**Comment by puranjayjain** [comment]
Feel free to reopen this issue if you have any problems.

**Comment by acomito** [comment]
Where do have you been putting it in projects using react-router? Do you need something like redux at that point or is the object available globally?

**Comment by puranjayjain** [comment]
It's not available globally you might need to pass that reference using an event emitter pattern or something equivalent, I haven't had the chance to explore redux at the moment.

**Comment by irrvrsl** [comment]
Same problem here: i can't get notifications. I've took code(`<ReactMaterialUiNotifications>` and method calling) from Main.js of the project and nothing happens.

**Comment by tiagowippel** [comment]
Hi, same problem here.. using React Router

**Comment by akashihi** [comment]
Same problem. No router, have <ReactMaterialUiNotifications/> on the page. Notifications are not shown.

**Comment by puranjayjain** [comment]
@tiagowippel @akashihi can you give me the link to the place in the project where it is being implemented if it is open source.

**Comment by akashihi** [comment]
@puranjayjain Component is defined here: https://github.com/akashihi/mdg-web-ui/blob/master/src/js/components/Notification.js and used here: https://github.com/akashihi/mdg-web-ui/blob/master/src/js/components/Main.js

Notifications are issued here: https://github.com/akashihi/mdg-web-ui/blob/master/src/js/util/ApiUtils.js#L12

**Comment by puranjayjain** [comment]
So @akashihi the problem seems to be that the show notifications is not being called in the same component the element is hoisted in. You can either move the showNotification call to Notification.js or create a ref to the notification.js's ReactMaterialUiNotifications component and trigger an action on that ref.

**Comment by akashihi** [comment]
@puranjayjain Well, i'm pretty new to the Js/React world. Could you please provide some intuition, how to implement that ref? Thanks in advance.

**Comment by DevAlien** [comment]
It does not work because you have to call a setState or forceUpdate on your own, if not the notifications will not be updated

**Comment by puranjayjain** [comment]
@DevAlien @akashihi this issue has been fixed in v0.5.0 and released on npm also!
